syntax = "proto3";
package securedexchange;

service Exchange {
    rpc SendModel (stream Model) returns (SimpleReply);
    rpc RunModel (stream Data) returns (ModelResult);
}

message Model {
    bytes data = 1;
    uint64 length = 2;
    repeated int32 input_fact = 3;
    int32 datum = 4;
}

message Data {
    bytes input = 1;
}

message ModelResult {
    repeated float output = 1;
    bool ok = 2;
    string msg = 3;
}

message SimpleReply {
    bool ok = 1;
    string msg = 2;
}